nrf-hal-common/src/spis.rs

```diff
@@ -0,0 +1,590 @@
+use core::{
+    ops::Deref,
+    sync::atomic::{compiler_fence, Ordering},
+};
+
+#[cfg(feature = "9160")]
+use crate::pac::{
+    spis0_ns::{
+        self as spis0, _EVENTS_ACQUIRED, _EVENTS_END, _EVENTS_ENDRX, _TASKS_ACQUIRE, _TASKS_RELEASE,
+    },
+    SPIS0_NS as SPIS0,
+};
+
+#[cfg(not(feature = "9160"))]
+use crate::pac::{
+    spis0::{self, _EVENTS_ACQUIRED, _EVENTS_END, _EVENTS_ENDRX, _TASKS_ACQUIRE, _TASKS_RELEASE},
+    SPIS0,
+};
+
+#[cfg(any(feature = "52832", feature = "52833", feature = "52840"))]
+use crate::pac::{SPIS1, SPIS2};
+
+use crate::{
+    gpio::{Floating, Input, Pin},
+    pac::{generic::Reg, Interrupt},
+    target_constants::{EASY_DMA_SIZE, SRAM_LOWER, SRAM_UPPER},
+};
+use embedded_dma::*;
+
+pub struct Spis<T: Instance>(T);
+
+impl<T> Spis<T>
+where
+    T: Instance,
+{
+    /// Takes ownership of the raw SPIS peripheral, returning a safe wrapper.
+    pub fn new(
+        spis: T,
+        sck_pin: &Pin<Input<Floating>>,
```

Comment:
No strong feeling, but we might want to make a `Pins` struct to match SPIM, etc.

Reply:
Oh wait. Stronger objection. I think `Spis` should own these pins. Even though it never uses them, it would be incorrect to allow the user to retain ownership of them and potentially modify the pin states.